theano/compile/function_module.py

```diff
@@ -751,7 +752,13 @@ def checkSV(sv_ori, sv_rpl):
         f_cpy.maker.fgraph.name = name
         return f_cpy
 
-    def __call__(self, *args, **kwargs):
+    def apply(self, inputs, outputs):
+        # TODO: assert that there's a list of outputs
+        self.output_idx = outputs # indirectly passing output indices
+        # assuming for now I pass all the inputs
+        return [self.__call__(*inputs)[i] for i in outputs]
```

Comment:
Do only 1 call to \__call__ otherwise, there will be duplicate computation of the common part. But this is a good direction.